functional/config.rb

```diff
@@ -0,0 +1,58 @@
+# Size of the CoreOS cluster created by Vagrant
```

Comment:
I am really struggling to agree that it's simpler to maintain the Vagrantfile in sync with that from coreos-vagrant - [_but not quite_](https://github.com/coreos/fleet/pull/1430/files#diff-a508e808c0fca80a337e9a8d22945e85R142) - rather than just port it to a much simpler Vagrantfile that fulfills our very particular purposes for testing here. 

In case it's not clear why I am pushing back on all these things: every file we add increases the complexity and burden of understanding and maintaining the codebase. I am asking lots of dumb questions here because even though I'm familiar with fleet and CoreOS it's still far from obvious to me why we really need all these parts, and e.g. where I should start making changes if I want to modify things. How can we make this easier for others approaching who have even less context?